WordPress/src/main/java/org/wordpress/android/util/CrashLogging.kt

```diff
@@ -0,0 +1,143 @@
+package org.wordpress.android.util
+
+import android.content.Context
+import android.preference.PreferenceManager
+import io.sentry.android.core.SentryAndroid
+import io.sentry.core.Sentry
+import io.sentry.core.protocol.User
+import org.wordpress.android.BuildConfig
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.store.AccountStore
+import org.wordpress.android.util.AppLog.T
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val EVENT_BUS_MODULE = "org.greenrobot.eventbus"
+private const val EVENT_BUS_EXCEPTION = "EventBusException"
+private const val EVENT_BUS_INVOKING_SUBSCRIBER_FAILED_ERROR = "Invoking subscriber failed"
+
+@Singleton
+class CrashLogging @Inject constructor(
+    private val accountStore: AccountStore
+) {
+    fun start(context: Context) {
+        SentryAndroid.init(context) { options ->
+            options.dsn = BuildConfig.SENTRY_DSN
+            options.cacheDirPath = context.cacheDir.absolutePath
+            options.isEnableSessionTracking = true // Release Health tracking
+            options.setBeforeSend { event, _ ->
+
+                if (!this.shouldSendEvents(context)) return@setBeforeSend null
+
+                if (event.exceptions.size > 1) {
+                    event.exceptions.lastOrNull()?.let { lastException ->
+                        // Remove the "Invoking subscriber failed" exception so that the main error will show up
+                        // in Sentry. This error only means that an exception occurred during an EventBus event and
+                        // it's not particularly useful for debugging.
+                        if (lastException.module == EVENT_BUS_MODULE &&
+                                lastException.type == EVENT_BUS_EXCEPTION &&
+                                lastException.value == EVENT_BUS_INVOKING_SUBSCRIBER_FAILED_ERROR) {
+                            event.exceptions.remove(lastException)
+                        }
+                    }
+                }
+                event
+            }
+        }
+
+        Sentry.setTag("version", BuildConfig.VERSION_NAME)
+
+        this.accountStore.account.apply {
```

Comment:
This doesn't deal with user account transitions. I thought about moving it to `beforeSend`, but the issue there is that it's called when Sentry starts up, and I'm not sure if the `accountStore` has the user data at that point, so it sort of becomes a hidden dependency. 

As noted in the PR description, the Tracks version of this code is a bit more resilient against this sort of issue, so the next step will be tackling that migration.

Reply:
Account transitions can be handled in a different way, but I am not sure if I understand why using `beforeSend` would be an issue. Since `beforeSend` is a callback, it should be called just before sending the error. Also, since `AccountStore` is a argument passed in the constructor it should be ready to use at any point in this class. Am I missing something?

Reply:
Chatted about this separately – because FluxC still reads from the DB on initialization, this should always be correct internally.

Changed in 7941de4843c6f1f1eaac1209466aa0b13eef21d8